api/v1/policy_types.go

```diff
@@ -8,132 +8,180 @@ import (
 	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
-// RemediationAction describes weather to enforce or inform
+// RemediationAction specifies the remediation of the policy. The parameter values are "enforce" and
+// "inform".
+//
 // +kubebuilder:validation:Enum=Inform;inform;Enforce;enforce
 type RemediationAction string
 
 const (
-	// Enforce is an remediationAction to make changes
 	Enforce RemediationAction = "Enforce"
-
-	// Inform is an remediationAction to only inform
-	Inform RemediationAction = "Inform"
+	Inform  RemediationAction = "Inform"
 )
 
-// PolicyTemplate template for custom security policy
+// PolicyTemplate is the definition of the policy engine resource to apply to the managed cluster,
+// along with configurations on how it should be applied.
 type PolicyTemplate struct {
-	// +kubebuilder:pruning:PreserveUnknownFields
 	// A Kubernetes object defining the policy to apply to a managed cluster
+	//
+	// +kubebuilder:pruning:PreserveUnknownFields
 	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
 
-	// Additional PolicyDependencies that only apply to this template
+	// ExtraDependencies is additional PolicyDependencies that only apply to this policy template.
+	// ExtraDependencies is a list of dependency objects detailed with extra considerations for
+	// compliance that should be fulfilled before applying the policy template to the managed
+	// clusters.
 	ExtraDependencies []PolicyDependency `json:"extraDependencies,omitempty"`
 
-	// Ignore this template's Pending status when calculating the overall Policy status
+	// IgnorePending is a boolean parameter to specify whether to ignore the "Pending" status of this
+	// template when calculating the overall policy status. The default value is "false" to not ignore a
+	// "Pending" status.
 	IgnorePending bool `json:"ignorePending,omitempty"`
 }
 
-// ComplianceState shows the state of enforcement
+// ComplianceState reports the observed status resulting from the definitions of the policy.
+//
+// +kubebuilder:validation:Enum=Compliant;Pending;NonCompliant
 type ComplianceState string
 
 const (
-	// Compliant is a ComplianceState
-	Compliant ComplianceState = "Compliant"
-
-	// NonCompliant is a ComplianceState
+	Compliant    ComplianceState = "Compliant"
 	NonCompliant ComplianceState = "NonCompliant"
-
-	// Pending is a ComplianceState
-	Pending ComplianceState = "Pending"
+	Pending      ComplianceState = "Pending"
 )
 
 // Each PolicyDependency defines an object reference which must be in a certain compliance
 // state before the policy should be created.
 type PolicyDependency struct {
 	metav1.TypeMeta `json:",inline"`
 
-	// The name of the object to be checked
+	// Name is the name of the dependent object to be checked.
```

Comment:
Is "dependent object" the correct term? To me it sounds like the object you reference here depends on this policy.

Maybe "depended on object" is the correct term?